### DISCUSSION TITLE
Auto reconnect

### DISCUSSION THREAD

**Discussion by thewh1teagle** [serialport/serialport-rs#150] [General]
How can I reconnect device after reconnect it to USB? (Windows / Linux)

**Comment by sirhcel** [comment]
How are you identifying your device in the first place? By its device file name?

If you can identify your device by VID, PID, manufacturer, product, serial number, ... then you could start watching for it reappearing by matching the `SerialPortInfo`s returned by [`available_ports`](https://docs.rs/serialport/4.3.0/serialport/fn.available_ports.html) periodically.

**Reply by thewh1teagle** [reply]
Yes.
See [serust](https://github.com/thewh1teagle/serust)

**Reply by sirhcel** [reply]
Great to hear that this solves automatic reconnects for your. Thank you for the pointer!
